jvb/src/main/kotlin/org/jitsi/videobridge/cc/allocation/Prioritize.kt

```diff
@@ -39,6 +41,23 @@ fun <T : MediaSourceContainer> prioritize(
     return conferenceEndpoints
 }
 
+fun prioritize2(
+    conferenceSources: MutableList<MediaSourceDesc>,
+    selectedSourceNames: List<String> = emptyList()
+): List<MediaSourceDesc> {
+    // Bump selected sources to the top of the list.
+    selectedSourceNames.asReversed().forEach { selectedSourceName ->
+        // Note the usage of sourceName!! which is expected to be always defined in the multi-stream mode
```

Comment:
I wonder if we should fill in missing source names in the bridge? Or make them mandatory in colibri?

Reply:
I think until we get there you should throw an exception earlier (e.g. at signaling time) if multi stream is enabled and a source doesn't have a name.

Reply:
With multi stream off, we can enforce:
1. At least one video source per endpoint
2. Set the source's name to the endpoint ID if missing

Reply:
I wonder if, once multi stream is finished and we're happy with how it works, we can substitude the duplicated code with source-name based code with sourceName=endpointId

Reply:
I will add a check to enforce source name on the Colibri level, but the `!!` should stay here regardless, right?

Reply:
Enforcing one video source per endpoint may not be a good idea for JVB usage outside of jitsi-meet? Especially not when the multi-stream is off as it would break backwards compatibility?

Reply:
I like the idea of substituting the duplicated code with source-name based code. Note that there's no backwards compatibility mode implemented in the multi-stream to support older mobile react-native clients yet, but this can be one way of solving it.

Reply:
I meant enforcing at *most* one video source per endpoint (not at least). Then we can make `MediaSourceDesc.sourceName` non-nullable (and polyfill with the endpoint ID in case of non-multi-stream), so you don't need the `!!`. But I don't want to block because of it.